### PR TITLE
Move invalidation of caches to ActionQueue

### PR DESCRIPTION
**Pull request by hazzik**

### CODE REVIEW

File: src/NHibernate/Cache/UpdateTimestampsCache.cs [hunk 1]
```diff
@@ -33,29 +33,40 @@ public UpdateTimestampsCache(Settings settings, IDictionary<string, string> prop
 			updateTimestamps = settings.CacheProvider.BuildCache(regionName, props);
 		}
 
+		public void PreInvalidate(object[] spaces)
```
Comment (gliljas) [inline src/NHibernate/Cache/UpdateTimestampsCache.cs:36]: Make obsolete?

File: src/NHibernate/Cache/UpdateTimestampsCache.cs [hunk 2]
```diff
@@ -33,29 +33,40 @@ public UpdateTimestampsCache(Settings settings, IDictionary<string, string> prop
...
 			{
-				updateTimestamps.Put(spaces[i], ts);
+				updateTimestamps.Put(space, ts);
 			}
+
 			//TODO: return new Lock(ts);
 		}
 
 		/// <summary></summary>
+		public void Invalidate(object[] spaces)
```
Comment (gliljas) [inline src/NHibernate/Cache/UpdateTimestampsCache.cs:55]: Make obsolete?

**Review by gliljas — COMMENTED** [review]
Ok, code is better than words :) Just a couple of potential obsoletions, but maybe that destroys it for asyncgenerator ?

**Comment by hazzik** [comment]
I was thinking to make them obsolete, but decided not to, I don’t know why.

**Comment by hazzik** [comment]
Maybe because of AsyncGenerator

**Comment by gliljas** [comment]
Let's leave them as is for now.

**Pull request merged**